plugin/keyring_vault/vault_io.cc

```diff
@@ -0,0 +1,136 @@
+#include <my_global.h>
+#include "vault_io.h"
+#include "vault_keys_list.h"
+#include "vault_credentials_parser.h"
+#include <curl/curl.h>
+#include <sstream>
+
+namespace keyring {
+
+my_bool Vault_io::init(std::string *keyring_storage_url)
+{
+  Vault_credentials_parser vault_credentials_parser(logger);
+  Vault_credentials vault_credentials;
+  return vault_credentials_parser.parse(keyring_storage_url, &vault_credentials) ||
+         vault_curl->init(&vault_credentials);
+}
+
+Vault_io::~Vault_io()
```

Comment:
It is responsible for lifetime of vault_curl and parser. Why it should not be. This way I am sure they will be deleted when Vault_io is no longer needed.